ray-operator/controllers/ray/common/service.go

```diff
@@ -317,6 +321,38 @@ func BuildHeadlessServiceForRayCluster(rayCluster rayv1.RayCluster) *corev1.Serv
 	return headlessService
 }
 
+// GetServePort finds the container port named "serve" in the RayCluster's head group spec.
+// It returns the default Ray Serve port 8000 if not explicitly defined.
+func GetServePort(cluster *rayv1.RayCluster) gwv1.PortNumber {
+	if cluster == nil || len(cluster.Spec.HeadGroupSpec.Template.Spec.Containers) == 0 {
+		return gwv1.PortNumber(utils.DefaultServingPort)
+	}
+
+	// Find the port named "serve" in the head group's container spec.
+	headContainer := cluster.Spec.HeadGroupSpec.Template.Spec.Containers[utils.RayContainerIndex]
+	for _, port := range headContainer.Ports {
+		if port.Name == utils.ServingPortName {
+			return gwv1.PortNumber(port.ContainerPort)
+		}
+	}
+
+	return gwv1.PortNumber(utils.DefaultServingPort)
+}
+
+// GetGatewayListenersForRayService constructs the default HTTP listener for a RayService Gateway.
+func GetGatewayListenersForRayService(rayServiceInstance *rayv1.RayService) []gwv1.Listener {
+	hostname := fmt.Sprintf("%s.%s.svc.cluster.local", rayServiceInstance.Name, rayServiceInstance.Namespace)
+
+	return []gwv1.Listener{
+		{
+			Name:     gwv1.SectionName(utils.GatewayListenerPortName),
+			Protocol: gwv1.HTTPProtocolType,
+			Port:     utils.DefaultGatewayListenerPort,
+			Hostname: (*gwv1.Hostname)(&hostname), // backwards compatibility with Serve service
```

Comment:
```suggestion
```

I think we still need to leave `Hostname` unspecified since we are not sharing the gateway with others. We don't need to require users to add `Host: xxx.svc.cluster.local` in their HTTP request headers.